store/store.go

```diff
@@ -812,6 +833,7 @@ func decodeJSONBody(resp *http.Response, success interface{}, failure interface{
 	if !ok {
 		result = failure
 	}
+	// XXX should we check content-type first?
```

Comment:
we could but we need to look at all usages, I don't remember if the store always uses application/json on JSON errors. If you look at assertion code it might be using a different special type

Reply:
Removed this comment, is now in the TODO per the comment below.